src/inmanta/deploy/state.py

```diff
@@ -324,6 +336,17 @@ def update_requires(
         """
         Update the requires relation for a resource. Updates the reverse relation accordingly.
         """
+        # If the resource is blocked transiently, and we drop at least one of its requirements
+        # we check to see if the resource can now be unblocked
+        # i.e. all of its dependencies are now compliant with the desired state.
+        if (
+            self.resource_state[resource].blocked is BlockedStatus.TRANSIENT
+            and self.requires[resource] - requires
+            and self.are_dependencies_compliant(resource)
```

Comment:
This is prior to the update of the requires, so it will include the removed ones, I think

Reply:
You are right. I updated this and added some more assertions to the test

Reply:
The fix is not necessarily correct either. I'm pretty sure the sets are mutable, so you're using the same set left and right of the `-` operator. I think you'll have to go for either:
- drop the method `are_dependencies_compliant` and inline it in both places because it's so simple
- two-stage it in this method: first check `dropped_requires: bool`, then update the requires, then check if dependencies are compliant and act on it.